### ISSUE TITLE
[Feature Request] Add organization collection `data` type and `resource` type

### ISSUE THREAD

**Issue by leynebe** [maxlaverse/terraform-provider-bitwarden#62]
Hello again!

I really like the provider and we use it a lot at my current project.
However, currently we have to hardcode a lot of the collection ID's in our Terraform code, we'd like to be able to either pull the collection ID via a `data` type or manage the collection via Terraform using the `resource` type.
I've not found a roadmap or anything but I assume you're planning on adding this functionality in the long run.
Is this something you're working on already? If not, I might have a go at it myself in my spare time, but I'm not very familiar with Go.

Thanks in advance.

**Comment by leynebe** [comment]
Now with some pseudocode:
```
data "bitwarden_organization_collection" "name" {
    name = "some_name" # Required
    parent_organization_collection_id = "xxx-yyy-zzz" # Optional
    organization_id = "xxxyyyzzz" # Required, dont know if this is necessary, but saw it in `bw` cli.
}

resource "bitwarden_organization_collection" "name" {
    name = "some_name" # Required
    parent_organization_collection_id = "xxx-yyy-zzz" # Optional, if null, on root level
    organization_id = "xxxyyyzzz" # Required, dont know if this is necessary, but saw it in `bw` cli.
}

resource "bitwarden_item_login" "name_data" {
  name            = "some_name_data"
  username        = "some_name_data"
  password        = "some_pass_data"
  collection_ids  = [data.bitwarden_organization_collection.name.id]
  organization_id = "xxxyyyzzz"
}

resource "bitwarden_item_login" "name_data_resource" {
  name            = "some_name_resource"
  username        = "some_name_resource"
  password        = "some_pass_resource"
  collection_ids  = [bitwarden_organization_collection.name.id]
  organization_id = "xxxyyyzzz"
}
```

**Comment by maxlaverse** [comment]
Hi @leynebe !
I see the org-collections can be created with the Bitwarden CLI, so it's definitely something that would be worth implementing in the provider. Your pseudo configuration looks good. I don't have that many collection which is why I've been regrouping them in terraform local variables so far.

I see you propose to find the data source by `name`. I think it's a very good idea, and I actually wanted to work on something similar but lacked time. Ideally, we would enable anyone to reference a datasouce by `name`, but the `id` would also be valid and take precedence. This much more friendly for the users, if they're fine with the uncertainty around name-uniqueness and modification.

**Comment by maxlaverse** [comment]
Hi @leynebe,
It's been a long time!
Yesterday I implemented search filters on _data sources_ as part of https://github.com/maxlaverse/terraform-provider-bitwarden/issues/123. This was half of the work required to implement this feature request, which is why I followed up with implementing `bitwarden_org_collection` and `bitwarden_organization` data sources. I wasn't sure about the naming, and ultimately I decided to stay close to Bitwarden's (`bitwarden_org_collection` instead of the proposed `bitwarden_organization_collection`).

Those changes are released in [v0.8.0](https://github.com/maxlaverse/terraform-provider-bitwarden/releases/tag/v0.8.0) and should allow to have mostly _"Bitwarden ID-free"_ Terraform workspaces.
